crates/fuzzing/src/generators/single_inst_module.rs

```diff
@@ -0,0 +1,307 @@
+    binary!(F32Sub, f32),
+    binary!(F64Sub, f64),
+    binary!(F32Mul, f32),
+    binary!(F64Mul, f64),
+    binary!(F32Div, f32),
+    binary!(F64Div, f64),
+    binary!(F32Min, f32),
+    binary!(F64Min, f64),
+    binary!(F32Max, f32),
+    binary!(F64Max, f64),
+    binary!(F32Copysign, f32),
+    binary!(F64Copysign, f64),
+    // Floating-point comparison.
+    compare!(F32Eq, f32),
+    compare!(F64Eq, f64),
+    compare!(F32Ne, f32),
+    compare!(F64Ne, f64),
+    compare!(F32Lt, f32),
+    compare!(F64Lt, f64),
+    compare!(F32Gt, f32),
+    compare!(F64Gt, f64),
+    compare!(F32Le, f32),
+    compare!(F64Le, f64),
+    compare!(F32Ge, f32),
+    compare!(F64Ge, f64),
+    // Integer conversions ("to integer").
+    unary!(I32Extend8S, i32),
+    unary!(I32Extend16S, i32),
+    unary!(I64Extend8S, i64),
+    unary!(I64Extend16S, i64),
+    convert!(I64Extend32S, i32 -> i64),
+    convert!(I32WrapI64, i64 -> i32),
+    convert!(I64ExtendI32S, i32 -> i64),
+    convert!(I64ExtendI32U, i32 -> i64),
+    convert!(I32TruncF32S, f32 -> i32),
+    convert!(I32TruncF32U, f32 -> i32),
+    convert!(I32TruncF64S, f64 -> i32),
+    convert!(I32TruncF64U, f64 -> i32),
+    convert!(I64TruncF32S, f32 -> i64),
+    convert!(I64TruncF32U, f32 -> i64),
+    convert!(I64TruncF64S, f64 -> i64),
+    convert!(I64TruncF64U, f64 -> i64),
+    convert!(I32TruncSatF32S, f32 -> i32),
+    convert!(I32TruncSatF32U, f32 -> i32),
+    convert!(I32TruncSatF64S, f64 -> i32),
+    convert!(I32TruncSatF64U, f64 -> i32),
+    convert!(I64TruncSatF32S, f32 -> i64),
+    convert!(I64TruncSatF32U, f32 -> i64),
+    convert!(I64TruncSatF64S, f64 -> i64),
+    convert!(I64TruncSatF64U, f64 -> i64),
+    convert!(I32ReinterpretF32, f32 -> i32),
+    convert!(I64ReinterpretF64, f64 -> i64),
+    // Floating-point conversions ("to float").
+    convert!(F32DemoteF64, f64 -> f32),
+    convert!(F64PromoteF32, f32 -> f64),
+    convert!(F32ConvertI32S, i32 -> f32),
+    convert!(F32ConvertI32U, i32 -> f32),
+    convert!(F32ConvertI64S, i64 -> f32),
+    convert!(F32ConvertI64U, i64 -> f32),
+    convert!(F64ConvertI32S, i32 -> f64),
+    convert!(F64ConvertI32U, i32 -> f64),
+    convert!(F64ConvertI64S, i64 -> f64),
+    convert!(F64ConvertI64U, i64 -> f64),
+    convert!(F32ReinterpretI32, i32 -> f32),
+    convert!(F64ReinterpretI64, i64 -> f64),
+];
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn sanity() {
+        let sut = SingleInstModule {
+            instruction: Instruction::I32Add,
+            parameters: &[ValType::I32, ValType::I32],
+            results: &[ValType::I32],
+        };
+        let wasm = sut.encode();
+        let wat = wasmprinter::print_bytes(wasm).unwrap();
+        assert_eq!(
+            wat,
+            r#"(module
+  (type (;0;) (func (param i32 i32) (result i32)))
+  (func (;0;) (type 0) (param i32 i32) (result i32)
+    local.get 0
+    local.get 1
+    i32.add
+  )
+  (export "test" (func 0))
+)"#
```

Comment:
One nice test to add here as well might be iterating over `super::INSTRUCTIONS` and verifying that everything which pops out is a valid module.